### PR TITLE
Bacpop 156 - Initial unit tests

### PR DESCRIPTION
**Pull request by absternator**
This PR contains the setup of unit testing and unit tests the current ui-rewrite branch.

### CODE REVIEW

**Comment by EmmaLRussell** [comment]
I thought you were going to break this up into sensible sized PRs?

File: app/client-v2/src/config.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 export const getApiUrl = () => {
   if (import.meta.env.MODE === "development") {
     return "http://localhost:4000";
-  } else {
-    return "https://${window.location.host}/api";
   }
+  if (import.meta.env.MODE === "test") {
+    return "";
+  }
```
Comment (EmmaLRussell) [inline app/client-v2/src/config.ts:7]: I thought most of the tests for urls were for `stringContaining` so do you need this? Or is it for the mock server part?

File: app/client-v2/src/layouts/AppNav.vue [hunk 2]
```diff
@@ -52,7 +52,7 @@ const toggle = (event: MouseEvent) => {
       </template>
       <template #end>
         <div class="flex gap-3">
-          <div><Button :icon="themeIcon" @click="toggleTheme" outlined /></div>
+          <div><Button aria-label="theme-switcher" :icon="themeIcon" @click="toggleTheme" outlined /></div>
```
Comment (EmmaLRussell) [inline app/client-v2/src/layouts/AppNav.vue:55]: `aria-label` values should be human-consumable shouldn't they? Typically read out by a screen reader, so punctuation maybe not an issue, but I'd expect this to be formatted like a label. 
```suggestion
          <div><Button aria-label="Switch theme" :icon="themeIcon" @click="toggleTheme" outlined /></div>
```

File: app/client-v2/src/views/HomeView.vue [hunk 3]
```diff
@@ -114,7 +115,7 @@ const onRowEditSave = async (event: DataTableRowEditSaveEvent) => {
             <div>
               <InputGroup>
                 <InputText placeholder="Create new Project" v-model="newProjectName" @keydown.enter="addProject" />
-                <Button icon="pi pi-plus-circle" @click="addProject" />
+                <Button aria-label="add-project" icon="pi pi-plus-circle" @click="addProject" />
```
Comment (EmmaLRussell) [inline app/client-v2/src/views/HomeView.vue:118]:
```suggestion
                <Button aria-label="New project" icon="pi pi-plus-circle" @click="addProject" />
```

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]
Great stuff!

The tests aren't actually running on gha because the branch isn't targeting main. I had the same issue recently. see [this comment](https://github.com/bacpop/beebop-deploy/pull/4#pullrequestreview-1895420406) for Rich's suggested approach. 

Also, on BuildKite, the image isn't being pushed because of:
```
Error parsing reference: "mrcide/beebop-server:bacpop-156/initial-unit-test" is not a valid repository/tag: invalid reference format
```
I think that's probably because of the slash in the branch name.

Need to modify codecov too I guess to make sure we get coverage for the new front end tests.

File: app/client-v2/e2e/vue.spec.ts [hunk 4]
```diff
@@ -1,8 +0,0 @@
-import { test, expect } from '@playwright/test';
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/e2e/vue.spec.ts:1]: was just the boiler plate one from create vue

File: app/client-v2/src/__tests__/composables/useTheme.spec.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import { DARK_THEME, LIGHT_THEME, useTheme } from "@/composables/useTheme";
+import { usePrimeVue } from "primevue/config";
+import { ref } from "vue";
+import { useStorage } from "@vueuse/core";
+
+// vitest.mock("@vueuse/core", () => ({
+//   useStorage: vitest.fn().mockReturnValue(ref(LIGHT_THEME))
+// }));
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/__tests__/composables/useTheme.spec.ts:8]: good catch :smile:

File: app/client-v2/src/config.ts (hunk 1 above)
Comment (absternator) [reply app/client-v2/src/config.ts:7]: yeah i did but just changed to make things simpler to match in general.. so future tests and stuff

**Pull request closed without merging**